lib/gst/player/gstplayer.c

```diff
@@ -1660,6 +2181,200 @@ gst_player_get_pipeline (GstPlayer * self)
   return val;
 }
 
+/**
+ * gst_player_get_media_info:
+ * @player: #GstPlayer instance
+ *
+ * Returns: (transfer full): Current media info instance.
+ *
+ * the caller should free it with g_object_unref()
+ */
+GstPlayerMediaInfo*
+gst_player_get_media_info (GstPlayer *self)
+{
+  GstPlayerMediaInfo *info;
+
+  g_mutex_lock (&self->priv->lock);
+  info = gst_player_media_info_copy (self->priv->media_info);
```

Comment:
No need for a copy, can just be a new reference. That's why we wanted it to be immutable among other things :)